src/rpcclient/rpcclient/capture_fd.py

```diff
@@ -0,0 +1,57 @@
+from typing import Optional
+
+from rpcclient.darwin.symbol import DarwinSymbol
+
+FD_SIZE = 4
+BUFFERSIZE = 0x10000
+
+
+class CaptureFD:
+    def __init__(self, client, fd: int) -> None:
+        self._client = client
+        self.fd: int = fd
+        self._backupfd: Optional[DarwinSymbol] = None
+        self._pipefd: Optional[DarwinSymbol] = None
+
+    def __enter__(self) -> 'CaptureFD':
+        with self._client.safe_malloc(FD_SIZE * 2) as pipefds:
+            pipefds.item_size = FD_SIZE
+            if 0 != self._client.symbols.pipe(pipefds):
+                self._client.raise_errno_exception('pipe failed')
+            self._backupfd = self._client.symbols.dup(self.fd)
+            if self._backupfd is None or 0 > self._backupfd:
```

Comment:
After this, `_backupfd` cannot be `None`. We can set it to `None` just before we raise the error